client/webserver/site/src/js/app.ts

```diff
@@ -396,6 +393,19 @@ export default class Application {
       Doc.show(page.noteList)
       this.ackNotes()
     })
+
+    // event handler for external links in notifications.  This is a delegated handler
+    // in order to avoid rebinding click handlers if bindUrlHandlers() were to be called
+    // after prepend*Element()
+    if (window.openUrl) {
```

Comment:
It's an optimization really, this way we don't need to bind a handler for every added item.  `#noteList` doesn't receive clicks on elements other than the intended targets.

Reply:
Binding to elements as we insert them is an established pattern, and it's hard to call it an optimization when we're essentially re-writing `bindUrlHandlers` instead of adding a single line in `prependListElement`.